src/commands/data/connectors/destroy.ts

```diff
@@ -11,14 +12,23 @@ export default class ConnectorsDestroy extends BaseCommand {
     },
   ]
 
+  static flags = {
+    confirm: flags.string({
+      description: 'confirms destroying the connector if passed in',
+      required: false,
+    }),
+  }
+
   static examples = [
     '$ heroku data:connectors:destroy gentle-connector-1234',
+    '$ heroku data:connectors:destroy gentle-connector-1234 --confirm gentle-connector-1234',
   ]
 
   async run() {
-    const {args} = this.parse(ConnectorsDestroy)
+    const {args, flags} = this.parse(ConnectorsDestroy)
     const connector = args.connector
 
+    await confirmConnector(connector, flags.confirm)
```

Comment:
We could suppress the backtrace by wrapping this in a try/catch block. The `await` should convert a Promise failure to an exception.